pkg/p2p/libp2p/peer.go

```diff
@@ -163,6 +164,26 @@ func (r *peerRegistry) overlay(peerID libp2ppeer.ID) (swarm.Address, bool) {
 	return overlay, found
 }
 
+func (r *peerRegistry) isConnected(peerID libp2ppeer.ID, remoteAddr ma.Multiaddr) (swarm.Address, bool) {
+	r.mu.RLock()
+	overlay, found := r.overlays[peerID]
```

Comment:
I would suggest to have a little less nesting. These suggestions are just opinionated cosmetics, the code as it is is fine also.

- check if `remoteAddr == nil` and return if it is before acquiring the lock
- acquire lock and defer unlock
- return if `!found`
- return if connections map lookup `!ok`
- return instead of break in the for loop

Reply:
The function that was used for checking if peer is connected was the one before, named `overlay`. From that one, I just added additional check inside.

I do not think we should remote just if `remoteAddr == nil` because the `peerID` is the first and main thing that is checked. Admittedly current solution does additionally check for remote address.

The lookup in connections map should never return `false` in the `found == true`, there just was no reason not to check it (just to prevent possible panic).

However, I do understand that changes may make things more clear.

Reply:
Thanks for the explanation and for the change @zbiljic.